lib/brainstem/dsl/configuration.rb

```diff
@@ -11,8 +11,13 @@ class Configuration
       # @params [Object] parent_configuration The parent configuration object
       #   which the new configuration object should use as a base.
       def initialize(parent_configuration = nil)
-        @parent_configuration = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
-        @storage = ActiveSupport::HashWithIndifferentAccess.new
+        @parent_configuration  = parent_configuration || ActiveSupport::HashWithIndifferentAccess.new
+        @storage               = ActiveSupport::HashWithIndifferentAccess.new
+
+        self.nonheritable_keys =
+          parent_configuration.respond_to?(:nonheritable_keys) &&
+          parent_configuration.nonheritable_keys ||
```

Comment:
Okay, so I've spent a bit more time on this. There is a bug here but I also think it's important that this be documented somewhere, so that future developers can understand the rationale behind programming the nonheritable keys this way. (I've also included it inside the configuration documentation.)

```ruby
        #
        # Nonheritable keys are a bit peculiar: they make the lookup for a key
        # specified as nonheritable to return no result when it falls back to
        # the parent configuration.
        #
        # These keys themselves are inheritable; in this way, a class that
        # descends from another will keep the same behaviour as its superclass
        # without necessarily having the same data. Or to put it another way,
        # if you have specified that 'title' is not inheritable in a
        # superclass's configuration, that is a property of that class, and
        # descendent classes should behave the same way.
        #
        # It is also unlikely that subclasses will modify the list of
        # nonheritable keys.
```

Here is the output with the fix:

```ruby
[1] pry(main)> parent = Brainstem::DSL::Configuration.new
=> #<Brainstem::DSL::Configuration:0x007fade41ffa10 @nonheritable_keys=#<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade41ff920 @parent_array=[], @storage=[]>, @parent_configuration={}, @storage={}>
[2] pry(main)> child = Brainstem::DSL::Configuration.new(parent)
=> #<Brainstem::DSL::Configuration:0x007fade41853f0
 @nonheritable_keys=#<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade4185378 @parent_array=#<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade41ff920 @parent_array=[], @storage=[]>, @storage=[]>,
 @parent_configuration=#<Brainstem::DSL::Configuration:0x007fade41ffa10 @nonheritable_keys=#<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade41ff920 @parent_array=[], @storage=[]>, @parent_configuration={}, @storage={}>,
 @storage={}>
[3] pry(main)> parent.nonheritable!(:a)
=> ["a"]
[4] pry(main)> child.nonheritable_keys
=> #<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade4185378 @parent_array=#<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade41ff920 @parent_array=[], @storage=["a"]>, @storage=[]>
[5] pry(main)> child.nonheritable_keys.to_a
=> ["a"]
[6] pry(main)> child.nonheritable!(:b)
=> ["b"]
[7] pry(main)> parent.nonheritable_keys
=> #<Brainstem::DSL::Configuration::InheritableAppendSet:0x007fade41ff920 @parent_array=[], @storage=["a"]>
[8] pry(main)> parent.nonheritable_keys.to_a
=> ["a"]
[9] pry(main)> child.nonheritable_keys.to_a
=> ["a", "b"]
```